pkg/apis/troubleshoot/v1beta2/supportbundle_types.go

```diff
@@ -27,6 +27,13 @@ type SupportBundleSpec struct {
 	HostCollectors  []*HostCollect     `json:"hostCollectors,omitempty" yaml:"hostCollectors,omitempty"`
 	Analyzers       []*Analyze         `json:"analyzers,omitempty" yaml:"analyzers,omitempty"`
 	HostAnalyzers   []*HostAnalyze     `json:"hostAnalyzers,omitempty" yaml:"hostAnalyzers,omitempty"`
+	// Uri defines the location of the spec file that needs to be used. This is optional and can be
+	// either a secret, url or spec yaml file.
+	// When presented with a spec document that contains a URI, if that spec is one of the specs provided
+	// on the command line or initial call to Troubleshoot, attempt to collect a replacement spec from that URI.
+	// If successful, it will replace the entire spec with the one downloaded.
+	// If unsuccessful, log the error and use the spec provided.
+	Uri string `json:"uri,omitempty" yaml:"uri,omitempty"`
```

Comment:
I'm curious the use case for anything other than a URL for this field. Is there a a reason to use a secret or yaml file in this reference?

This nuance may not matter for this actual change, which if true this comment is overly verbose I think. I like documenting in code, but if the comment documents things that aren't implemented here it runs the risk of being incorrect when other code changes and that change wouldn't know to come update this comment.

To me, I would simply include "URI optionally defines a location which is the source of this spec to allow updating of the spec at runtime".

Reply:
Agree with Chris' comments above. Also for:

> If successful, it will replace the entire spec with the one downloaded.

I didn't know this was our intention?

Reply:
Regards why we might use a secret, yaml file, or URL with an oci:// type, it's possible that these could be handy but far from essential.  The reality is that the simplest way to use the code to ingest the content of the URI could be to use the same logic that's already there, which includes these types.

I do agree that it would be good to be less specific here with the type of location - the code change to implement use of that will dictate the end result though it looks currently like the simplest change is to accept any type of URI.

> > If successful, it will replace the entire spec with the one downloaded.
> 
> I didn't know this was our intention?

yup, that's the intent - see https://github.com/replicatedhq/troubleshoot/blob/main/design/proposal-updateable-specs.md

Reply:
> I'm curious the use case for anything other than a URL for this field. Is there a a reason to use a secret or yaml file in this reference?
> 
> This nuance may not matter for this actual change, which if true this comment is overly verbose I think. I like documenting in code, but if the comment documents things that aren't implemented here it runs the risk of being incorrect when other code changes and that change wouldn't know to come update this comment.
> 
> To me, I would simply include "URI optionally defines a location which is the source of this spec to allow updating of the spec at runtime".

Thanks @chris-sanders, it makes perfect sense indeed to minimize the explanation and clarify a comment line into just one sentence for this new field.

Reply:
> Regards why we might use a secret, yaml file, or URL with an oci:// type, it's possible that these could be handy but far from essential. The reality is that the simplest way to use the code to ingest the content of the URI could be to use the same logic that's already there, which includes these types.
> 
> I do agree that it would be good to be less specific here with the type of location - the code change to implement use of that will dictate the end result though it looks currently like the simplest change is to accept any type of URI.
> 
> > > If successful, it will replace the entire spec with the one downloaded.
> > 
> > 
> > I didn't know this was our intention?
> 
> yup, that's the intent - see https://github.com/replicatedhq/troubleshoot/blob/main/design/proposal-updateable-specs.md

Thanks @xavpaice for your feedback!

Reply:
> Agree with Chris' comments above. Also for:
> 
> > If successful, it will replace the entire spec with the one downloaded.
> 
> I didn't know this was our intention?

Thanks @diamonwiggins for your feedback!